src/controllers/user.js

```diff
@@ -138,6 +128,37 @@ const updatePassword = async (req, res) => {
   }
 };
 
+const updateAddress = async (req, res) => {
+  const User = await UserModel.findOne({ _id: req.user.userId });
+  try {
+    const { addressID, operation } = req.params;
+    const newAddress = req.body.address;
+    let message;
+    if (addressID && operation) {
+      const toUpdateAddress = User.address.filter(
+        (address) => String(address._id) === addressID
+      );
+      const index = User.address.indexOf(toUpdateAddress[0]);
+      if (index === -1)
+        return res.status(404).json({ message: 'Address not found' });
+      if (operation === 'update') {
```

Comment:
I tried to update a current address, and here's the outcome:
- This was the address we already had:
![Screenshot from 2022-08-18 17-45-48](https://user-images.githubusercontent.com/102853969/185428473-58a6356f-9fca-46c1-96a4-d4765c6c37dd.png)
- I tried to change it, and this is what I sent:
![Screenshot from 2022-08-18 17-45-51](https://user-images.githubusercontent.com/102853969/185428761-e32be6a9-70bc-4155-9cd2-387c89e2699d.png)
- I received a "address added," but it was added as a `null`, let alone the fact that it should've been an update.
- And this is what one receives when sends in the example:
![Screenshot from 2022-08-18 17-47-41](https://user-images.githubusercontent.com/102853969/185429164-bf39713b-e9be-4a1c-adb7-dc5aecbcaef3.png)